app/main.py

```
@@ -1,3 +1,81 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+        else:
+            return f"the {other} is not a Distance and it doesn't a number!"
+
+    def __iadd__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+            return self
+        elif isinstance(other, (int, float)):
+            self.km += other
+            return self
+        else:
+            return f"the {other} is not a Distance and it doesn't a number!"
```

Issue: Returning a string from __iadd__ when the type is unsupported is not standard for operator overloading.
You should raise a TypeError instead, to properly signal an unsupported operation.